src/components/swapv2/LimitOrder/EditOrderModal.tsx

```diff
@@ -0,0 +1,97 @@
+import { Trans } from '@lingui/macro'
+import { X } from 'react-feather'
+import { Flex, Text } from 'rebass'
+import styled from 'styled-components'
+
+import Modal from 'components/Modal'
+import { Z_INDEXS } from 'constants/styles'
+import { useCurrencyV2 } from 'hooks/Tokens'
+import { TransactionFlowState } from 'types'
+
+import LimitOrderForm, { Label } from './LimitOrderForm'
+import { calcInvert, calcPercentFilledOrder, calcRate, removeTrailingZero, uint256ToFraction } from './helpers'
+import { LimitOrder, LimitOrderStatus, RateInfo } from './type'
+
+const Wrapper = styled.div`
+  width: 100%;
+  padding: 20px 22px;
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+`
+
+const StyledLabel = styled(Label)`
+  margin-bottom: 0;
+`
+export default function EditOrderModal({
+  onCancelOrder,
+  onDismiss,
+  order,
+  note,
+  isOpen,
+  flowState,
+  setFlowState,
+  refreshListOrder,
+}: {
+  onCancelOrder: () => Promise<any>
+  onDismiss: () => void
+  order: LimitOrder
+  note: string
+  isOpen: boolean
+  flowState: TransactionFlowState
+  setFlowState: React.Dispatch<React.SetStateAction<TransactionFlowState>>
+  refreshListOrder: () => void
+}) {
+  const { status, makingAmount, takingAmount, makerAsset, takerAsset, filledTakingAmount, expiredAt } = order
+  const currencyIn = useCurrencyV2(makerAsset) ?? undefined
+  const currencyOut = useCurrencyV2(takerAsset) ?? undefined
+  const inputAmount = uint256ToFraction(makingAmount).toFixed(currencyIn?.decimals ?? 16)
+  const outputAmount = uint256ToFraction(takingAmount).toFixed(currencyOut?.decimals ?? 16)
```

Comment:
fallback `decimals` for 16? Is this mistake or intented

Reply:
why not use formatUnits from ethers here for amount display?

Reply:
const inputAmount = formatUnits(makingAmount, currencyIn.decimals ?? 18); 
it shorter and even no need to clear trailing zero.

Reply:
updated using formatUnits. and remove fallbaclk decimal 16.
But still need remove trailing zero.
1 => formatUnits will result 1.0 lol